### DISCUSSION TITLE
how to use this in vite+ react route?

### DISCUSSION THREAD

**Discussion by jesse996** [kiliman/remix-flat-routes#111] [General]

**Comment by mwondhaf** [comment]
In the vite.config.ts file, I put it under the vitePlugin in the plugins... see below my code
```
import { vitePlugin as remix } from "@remix-run/dev";
import { installGlobals } from "@remix-run/node";
import { flatRoutes } from "remix-flat-routes";
import { defineConfig } from "vite";
import tsconfigPaths from "vite-tsconfig-paths";

installGlobals();

export default defineConfig({
  plugins: [
    remix({
      ignoredRouteFiles: ["**/.*"],
      routes: async (defineRoutes) => {
        return flatRoutes("routes", defineRoutes);
      },
    }),
    tsconfigPaths(),
  ],
});

**Reply by makrandgupta** [reply]
@kiliman could we add this to the readme?

**Reply by kiliman** [reply]
Yeah, I plan on rewriting the README.

**Reply by leonardo-ornelas** [reply]
Does your CLI command show all routes?

`npx remix routes`

Here it shows almost nothing after updating to use vite, but routes work at runtime.

**Reply by leonardo-ornelas** [reply]
nvm, i just updated to the last remix and worked.

**Reply by kiliman** [reply]
Yes, prior to v2.8.1, the `remix` CLI didn't read the _vite.config.ts_ file, so you still needed a _remix.config.js_ file. Now you can keep everything in the Vite config.

**Reply by jesse996** [reply]
soory but i mean is it possible to use this with react router +vite , not remix +vite?

**Reply by kiliman** [reply]
@jesse996 React Router doesn't use file-based routes, so `remix-flat-routes` doesn't apply.
